test/integration/targets/vault/script/vault-secret.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -eu
+
+# shellcheck disable=SC2086
+basename="$(basename $0)"
+# shellcheck disable=SC2046
+# shellcheck disable=SC2086
+dirname="$(basename $(dirname $0))"
+basename_prefix="get-password"
+default_password="foo-bar"
+
+case "${basename}" in
+  "${basename_prefix}"-*)
+    password="${default_password}-${basename#${basename_prefix}-}"
```

Comment:
It is rather hard to parse what's going on here. Have you considered using Python and not shell for this script? I bet it'd be more readable.

Reply:
used the user supplied example, its just using the basename of the path and it's directory to generate the secret